File: backend/src/main/java/com/revnu/backend/features/auth/dto/RegisterRequest.java
package com.revnu.backend.features.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record RegisterRequest(
    @NotBlank @Email String email,
    @NotBlank
    @Pattern(
        regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[^A-Za-z\\d]).{8,}$",
        message = "Password must be at least 8 characters and include uppercase, lowercase, a number, and a special character."
    )
    String password,
    @NotBlank String fullname
) {}

File: mobile/app/src/main/java/com/revnu/mobile/features/auth/model/RegisterRequest.kt
package com.revnu.mobile.features.auth.model

import com.google.gson.annotations.SerializedName

data class RegisterRequest(
    @SerializedName("email")
    val email: String,

    @SerializedName("password")
    val password: String,

    @SerializedName("fullname")
    val restaurantName: String
)